packages/relay/src/lib/eth.ts

```diff
@@ -524,6 +524,12 @@ export class EthImpl implements Eth {
           return predefined.INVALID_PARAMETER(0, `Invalid 'value' field in transaction param. Value must be greater than 0`);
         }
       } else {
+        // The size limit of the encoded contract posted to the mirror node can cause contract deployment transactions to fail with a 400 response code.
+        // The contract is actually deployed on the consensus node, so the contract will work.  In these cases, we don't want to return a 
+        // CONTRTACT_REVERT error.
+        if (e.isContractReverted() && e.detail && e.data && e.message !== MirrorNodeClientError.messages.INVALID_HEX && process.env.ALLOW_INVALID_HEX_RESPONSE) {
```

Comment:
Q: why do we need to check for `e.detail && e.data`?

Reply:
I wanted to lock in all of the conditions that would describe the invalid hex error.  It is the exceptional case now.  So far we've treated 400 responses as CONTRACT_REVERT.

Reply:
I also not sure why we need to check for `e.detail && e.data`

Reply:
Removed.